cocos/core/scene-graph/node.ts

```diff
@@ -390,6 +391,16 @@ export class Node extends BaseNode {
 
     public _onBatchCreated (dontSyncChildPrefab: boolean) {
         super._onBatchCreated(dontSyncChildPrefab);
+
+        if (!dontSyncChildPrefab && this._prefabInstance) {
```

Comment:
`doNotSyncChildPrefab` ?

Reply:
Yes, here we merged piror _onBatchRestored to _onBatchCreated.
When soft reloading the scene in editor, the scene's children already contain instantiated prefab nodes (in serialized data), which can not be refreshed from the prefab asset again.
Same as instantiating a node in runtime/editor.

Reply:
So, we use a dontSyncChildPrefab to indicate whether we need to sync prefab data when initializing a node.